pyof/foundation/basic_types.py

```diff
@@ -421,10 +421,23 @@ def __init__(self, value=b''):  # noqa
         Raises:
             ValueError: If given value is not bytes.
         """
-        if not isinstance(value, bytes):
-            raise ValueError('BinaryData must contain bytes.')
+        value = self._pack_if_necessary(value)
         super().__init__(value)
 
+    @staticmethod
+    def _pack_if_necessary(value):
```

Comment:
I believe the `_if_necessary` suffix is an implementation detail that shouldn't be part of the method's name in this case. `_pack(value)` together with `@staticmethod` is enough for understanding its functionality.